src/main/connections/postgres/PostgresConnection.ts

```diff
@@ -16,45 +16,39 @@ export class PostgresConnection extends BaseConnection {
         );
     }
 
-    async define(text: string): Promise<void> {
-        await this.execute(text);
-    }
-
-    async modify(text: string, params?: any[]): Promise<SqlModificationResult> {
-        const res = await this.execute(text, params);
-
-        return {
-            affectedRowCount: res.rowCount ?? 0,
-            rows: res.rows,
-            fieldData: this.getFieldData(res.fields)
-        };
-    }
-
     async query(text: string, params?: any[]): Promise<SqlQueryResult> {
-        const res = await this.execute(text, params);
-
-        return {
-            rows: res.rows,
-            rowCount: res.rowCount ?? 0,
-            fieldData: this.getFieldData(res.fields)
-        };
-    }
-
-    release() {
-        this.client.release();
-    }
-
-    private async execute(text: string, params?: any[]) {
         try {
-            return await this.client.query(text, params);
+            const res = await this.client.query(text, params);
+
+            if (res.rows.length && res.rowCount) {
```

Comment:
Does it always have `res.rows`? (just curious as I don't know what the result type is)

Reply:
yeah, with `pg` the return type of the client is consistent, so in the case of DDL queries it's always an empty array, a rowCount of 0 and empty field data. So figured theres just no point in returning anything in that case.

`mysql2` is much more particular/specific about typing and return types which is why the code is a bit more involved in that one